packages/app/src/HomePage.utils.ts

```diff
@@ -9,6 +9,11 @@ import {
 } from '@medplum/core';
 import { Bundle, ResourceType, UserConfiguration } from '@medplum/fhirtypes';
 
+/** Custom navigation paths when the user clicks New... */
+export const RESOURCE_TYPE_CREATION_PATHS: Partial<Record<ResourceType, string>> = {
```

Comment:
I haven't seen this TS pattern before: `Partial<Record<K, V>>` — what does `Partial` do here?

Reply:
It makes all keys (i.e. ResourceTypes) optional in the object while still providing the type safety that every key specified is indeed a `ResourceType` to prevent you from fat-fingering `Patent` instead of `Patient`.

In the parts of the code I've come across so far, it looks like we tend to do `Record<string, ...>` when only a subset of resource types need to be included which doesn't offer the same type safety.

It comes at the expense of not being able to use `RESOURCE_TYPE_CREATION_PATHS.Bot` without typescript complaining, but that's arguably could be for the best depending on the situation.

Reply:
Cool, thanks for the explanation! 🙌

Reply:
Correct me if wrong, but I think this may have worked without `Partial` right now.  We have an open ticket to look into the `noUncheckedIndexedAccess` compiler flag, which would tighten that up. 
 https://github.com/medplum/medplum/issues/3009